core/Scheduler/Task.php

```diff
@@ -72,22 +83,31 @@ class Task
      *                                          should be executed and how long before the next execution.
      * @param int $priority The priority of the task. Tasks with a higher priority will be executed first.
      *                      Tasks with low priority will be executed last.
+     * @param int $ttlInSeconds TTL to use for this task. Defauts to 3600. See {@link self::$ttlInSeconds}
      * @throws Exception
      */
     public function __construct($objectInstance, $methodName, $methodParameter, $scheduledTime,
-                                $priority = self::NORMAL_PRIORITY)
+                                $priority = self::NORMAL_PRIORITY, int $ttlInSeconds = null)
```

Comment:
Should we put the default as the default value here and remove `$ttlInSeconds = $ttlInSeconds ?? 3600;` below? Is there any benefit in having it `null` here and later set it to the default if not set?

```suggestion
                                $priority = self::NORMAL_PRIORITY, int $ttlInSeconds = 3600)
```

Reply:
Had it that way in the first place. But that would require to also set this default value for the methods in `core/Plugin/Tasks.php` (e.g. hourly, weekly, custom,...), as passing `null` wouldn't be possible there anymore.

Reply:
Why wouldn't it be possible to pass null? We can make the type nullable `?int $ttlInSeconds = 3600` and explicitly passing in null would be more expressive than not passing it at all.. but no strong preference here.

Reply:
You could allow passing null. But then you wouldn't have a ttl set, which is required. So you would need the current fallback check again, or would need to ensure a (non null) value is passed from the other methods.

Reply:
Circular problem...